File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/entity/ExchangeService.java
package ru.nikitazhelonkin.coinbalance.data.entity;


import android.support.annotation.DrawableRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import ru.nikitazhelonkin.coinbalance.R;

public enum  ExchangeService {

    BITFINEX("Bitfinex", R.drawable.bitfinex),
    BITTREX("Bittrex", R.drawable.bittrex),
    BINANCE("Binance", R.drawable.binance),
    COINBASE("Coinbase", R.drawable.coinbase),
    GEMINI("Gemini", R.drawable.gemini),
    HITBTC("HitBTC", R.drawable.hitbtc),
    KRAKEN("Kraken", R.drawable.kraken),
    POLONIEX("Poloniex", R.drawable.poloniex);

    String name;
    int iconResId;

    ExchangeService(@NonNull String name, @DrawableRes int iconResId) {
        this.name = name;
        this.iconResId = iconResId;
    }

    @NonNull
    public String getName() {
        return name;
    }

    public int getIconResId() {
        return iconResId;
    }

    @Nullable
    public static ExchangeService forName(String name) {
        for (ExchangeService service : values()) {
            if (service.getName().equalsIgnoreCase(name)) {
                return service;
            }
        }
        return null;
    }

    @Override
    public String toString() {
        return getName();
    }
}

File: app/src/main/java/ru/nikitazhelonkin/coinbalance/data/entity/ExchangeService.kt
package ru.nikitazhelonkin.coinbalance.data.entity


import android.support.annotation.DrawableRes

import ru.nikitazhelonkin.coinbalance.R

enum class ExchangeService constructor(val title: String, @field:DrawableRes val iconResId: Int) {

    BITFINEX("Bitfinex", R.drawable.bitfinex),
    BITTREX("Bittrex", R.drawable.bittrex),
    BINANCE("Binance", R.drawable.binance),
    COINBASE("Coinbase", R.drawable.coinbase),
    GEMINI("Gemini", R.drawable.gemini),
    HITBTC("HitBTC", R.drawable.hitbtc),
    KRAKEN("Kraken", R.drawable.kraken),
    POLONIEX("Poloniex", R.drawable.poloniex),
    YOBIT("YoBit", R.drawable.yobit),
    WEX("Wex", R.drawable.wex);

    override fun toString(): String {
        return title
    }

    companion object {

        @JvmStatic fun forName(name: String): ExchangeService? {
            for (service in values()) {
                if (service.title.equals(name, ignoreCase = true)) {
                    return service
                }
            }
            return null
        }
    }
}
